lib/packages/core/lib/src/widgets/chatroom/chatroom_bar.dart

```diff
@@ -206,30 +228,35 @@ class _LMChatroomBarState extends State<LMChatroomBar> {
                 return Container(color: Colors.red);
               }),
             ),
-            Container(
-              width: double.infinity,
-              padding: EdgeInsets.only(
-                left: 2.w,
-                right: 2.w,
-                top: 1.5.h,
-                bottom: 1.5.h,
-              ),
-              child: Row(
-                mainAxisAlignment: MainAxisAlignment.spaceAround,
-                crossAxisAlignment: CrossAxisAlignment.center,
-                children: [
-                  _isRespondingAllowed()
-                      ? _defTextField(context)
-                      : _defDisabledTextField(context),
-                  if (_isRespondingAllowed())
-                    _screenBuilder.sendButton(
-                      context,
-                      _textEditingController,
-                      _onSend,
-                      _defSendButton(context),
-                    ),
-                ],
-              ),
+            ValueListenableBuilder(
```

Comment:
Do we need to wrap it in `ValueListenableBuider`, Does BlocBuilder not able to rebuild it after state change? as i am seeing that you are rebuilding it on `LMChatRefreshBarState`.

Reply:
No this is because, we are not rebuilding based on BLoC states. We are only listening for the state. And returning a default view only for all views.

Reply:
But BlocBuilder do rebuild every time when state gets changed. we are not checking for the state but we do check if member have permission to message on not to build The ChatroomBar and i think the listner is changing the permission state and if it do in same way then it should rebuild the correct bar. please confirm this once through debugger.

Reply:
We are not checking for the state in the builder like I said, this is because the majority of the things have to remain the same and we only have to handle hiding the send button and changing the hint text. If we do it using BLoC's builder then we will have to return a complete new widget for all those changed states. 

And I would've done it but I didn't want to mess with the reply and edit functionality already present, which would've resulted in extra bugs coming up in completed modules.